### ISSUE TITLE
Re-adding folder doesn't get reflected in VS Code until focus

### ISSUE THREAD

**Issue by TylerLeonhardt** [microsoft/vscode#134144]
Testing #133753

macOS. I have a VS Code window open to the `vscode` repo but it does NOT have focus. I instead have focus on another window of VS Code (or another app... same behavior)

Here's the script I'm running outside of VS Code:

All it does is renames the vscode folder that I have open in VS Code and renames it back 100 times.
```ps
1..100 | % {
    mv ./vscode ./vscode-olddd
    sleep 10
    mv ./vscode-olddd ./vscode
    sleep 10
}
```

I noticed that the _removal_ of the vscode folder gets reflected in the VS Code instance with the `vscode` repo (where it gives you the `!` that says this folder doesnt exist)

However, when the rename gets changed _back_ to `vscode` thus "recreating" the `vscode` folder, nothing is reflected in the VS Code instance with the `vscode` repo.

However, once I focus the VS Code instance with the `vscode` repo then the contents are refreshed and it shows up and everything is fine.

**Comment by TylerLeonhardt** [comment]
This same behavior happens when you rename the target of a symbolic link and then rename it back.

**Comment by bpasero** [comment]
Yeah I noticed this as well and wonder if this is an issue in the explorer itself because doing the same (`mv folderNameA folderNameB`) is nicely reflected in the explorer for children of the root.

We must be having some handling of change events for folders, but probably not for the root itself? I think if the root changes we should do a refresh of the tree, but probably only in this case where we had a deleted root before because we do not really want the entire tree to refresh in other cases for performance reasons.
